src/nvim/tui/tui.c

```diff
@@ -434,6 +433,83 @@ static void tui_cursor_goto(UI *ui, int row, int col)
   unibi_goto(ui, row, col);
 }
 
+static cursorentry_T decode_cursor_entry(Dictionary args)
+{
+  cursorentry_T r;
+
+  for (size_t i = 0; i < args.size; i++) {
+    char *keyStr = args.items[i].key.data;
+    Object value = args.items[i].value;
+
+    if (strcmp(keyStr, "cursor_shape") == 0) {
+      const char *shapeStr = args.items[i].value.data.string.data;
+      if (strcmp(shapeStr, "block") == 0) {
+        r.shape = SHAPE_BLOCK;
+      } else if (strcmp(shapeStr, "vertical") == 0) {
+        r.shape = SHAPE_VER;
+      } else if (strcmp(shapeStr, "horizontal") == 0) {
+        r.shape = SHAPE_HOR;
+      } else {
+        WLOG("Unknown value %s", shapeStr);
+      }
+    } else if (strcmp(keyStr, "blinkon") == 0) {
+      r.blinkon = value.data.integer;
+    } else if (strcmp(keyStr, "blinkoff") == 0) {
+      r.blinkoff = value.data.integer;
+    } else if (!strcmp(keyStr, "hl_id")) {
+      r.id = (int)value.data.integer;
+    }
+  }
+  return r;
+}
+
+static void tui_cursor_style_set(UI *ui, Dictionary args)
+{
+  TUIData *data = ui->data;
+  static char *modes[] = {
+    "normal",
```

Comment:
It's just a way to map string to integer. Moving it to cursor_shape.c would be interesting only as a function int str_to_int_mode(char*) other UIs could call.